receiver/awscontainerinsightreceiver/internal/host/ec2tags.go

```diff
@@ -0,0 +1,141 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package host
+
+import (
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"go.uber.org/zap"
+)
+
+const (
+	ClusterNameKey          = "container-insight-eks-cluster-name"
+	ClusterNameTagKeyPrefix = "kubernetes.io/cluster/"
+	AutoScalingGroupNameTag = "aws:autoscaling:groupName"
+)
+
+type Ec2Tags struct {
+	refreshInterval      time.Duration
+	instanceID           string
+	clusterName          string
+	autoScalingGroupName string
+	logger               *zap.Logger
+	shutdownC            chan bool
+}
+
+func NewEc2Tags(instanceID string, refreshInterval time.Duration, logger *zap.Logger) *Ec2Tags {
+	if instanceID == "" {
+		return nil
+	}
+
+	et := &Ec2Tags{
+		instanceID:      instanceID,
+		refreshInterval: refreshInterval,
+		shutdownC:       make(chan bool),
+		logger:          logger,
+	}
+
+	et.refresh()
+	go func() {
+		refreshTicker := time.NewTicker(et.refreshInterval)
+		defer refreshTicker.Stop()
+		for {
+			select {
+			case <-refreshTicker.C:
+				if et.clusterName != "" {
+					//stop once we get the cluster name
+					return
+				}
+				et.refresh()
+			case <-et.shutdownC:
+				return
+			}
+		}
+	}()
+
+	return et
+}
+
+func (et *Ec2Tags) fetchEc2Tags() map[string]string {
+	et.logger.Info("Fetch ec2 tags to detect cluster name and auto scaling group name")
+	tags := make(map[string]string)
+	//add some sleep jitter to prevent a large number of receivers calling the ec2 api at the same time
+	time.Sleep(hostJitter(3 * time.Second))
```

Comment:
have a `hostDefaultJitter()` to avoid passing the hard coded 3s here?

Reply:
I don't see a strong reason for creating another function.